src/briefcase/integrations/subprocess.py

```diff
@@ -26,6 +31,53 @@ def json_parser(json_output):
         raise ParseError(f"Failed to parse output as JSON: {e}")
 
 
+def is_process_dead(pid: int):
+    """
+    Returns True if a PID is not assigned to a process.
+
+    Checking if a PID exists is only a semi-safe proxy to determine
+    if a process is dead since PIDs can be re-used. Therefore, this
+    function should only be used via constant monitoring of a PID
+    to identify when the process goes from existing to not existing.
+
+    :param pid: integer value to be checked if assigned as a PID.
+    :return: True if PID doesn't exist; defaults to False if PID=None.
+    """
+    if pid is None:
+        return False
+    return not psutil.pid_exists(pid)
+
+
+def get_process_id_by_command(command_list: list = None, command: str = ""):
+    """
+    Find a Process ID (PID) a by its command.
+    If multiple processes are found, then the most recently created
+    process ID is returned.
+
+    :param command_list: list of a command's fully qualified path and its arguments.
+    :param command: a partial or complete fully-qualified filepath to a command.
+        This is primarily intended for use on macOS where the `open` command
+        takes a filepath to a directory for an application; therefore, the actual
+        running process will be running a command within that directory.
+    :return: PID if found else None
+    """
+    matching_processes = []
+    for proc in psutil.process_iter(["cmdline", "create_time", "pid"]):
+        proc_cmdline = proc.info['cmdline']
+        if command_list and proc_cmdline == command_list:
+            matching_processes.append(proc.info)
+        if command and proc_cmdline and proc_cmdline[0].startswith(command):
+            matching_processes.append(proc.info)
+
+    if len(matching_processes) == 1:
+        return matching_processes[0]["pid"]
+    elif len(matching_processes) > 1:
+        # return the ID of the most recently created matching process
+        return sorted(matching_processes, key=operator.itemgetter("create_time"))[-1]["pid"]
```

Comment:
Given that this is an explicitly ambiguous case, it's probably worth logging that we *might* have found the wrong process. I suspect this will be a very unlikely edge case - but then, one in a million chances happen nine times out of ten :-) On that basis, it's worth flagging to avoid any confusion when it (inevitably) happens.

Reply:
Makes sense. Although, access to a `logger` is not immediately available here. This hearkens back a bit to making `Console` and/or `Log` singletons so their universal access would be trivial.

In lieu of that, I simply added `logger` as an optional parameter to this function. I'm not particularly keen on this...but the other alternative I see is to pigeonhole this function inside `Subprocess` which I think I'm even less keen on.

Thoughts?

P.S. While this function is general in nature, the message I added to be logged does imply the process being sought is the user's Application. I think this compromise makes sense, though, given the nature of this project.